File: attendence.py
from datetime import datetime

class AttendanceSystem:
    def __init__(self):
        self.attendance_list = {}

    def mark_attendance(self, name):
        date = datetime.now().strftime("%Y-%m-%d")
        
        if name in self.attendance_list:
            if date in self.attendance_list[name]:
                print(f"Attendance already marked for {name} on {date}.")
            else:
                self.attendance_list[name].append(date)
                print(f"{name} has been marked present for {date}.")
        else:
            self.attendance_list[name] = [date]
            print(f"{name} has been marked present for {date}.")

    def view_attendance(self):
        if not self.attendance_list:
            print("No attendance records found.")
        else:
            for name, dates in self.attendance_list.items():
                print(f"{name}: {', '.join(dates)}")

    def view_specific_attendance(self, name):
        if name in self.attendance_list:
            dates = self.attendance_list[name]
            print(f"{name} has been present on: {', '.join(dates)}")
        else:
            print(f"No attendance records found for {name}.")


class AttendanceManagement:
    def __init__(self):
        self.attendance_system = AttendanceSystem()

    def run(self):
        print("Attendance Management System")
        
        while True:
            print("\nOptions:")
            print("1. Mark Attendance")
            print("2. View Attendance")
            print("3. View Specific Attendance")
            print("4. Exit")

            option = input("Select an option: ")
            if option == "1":
                name = input("Enter name: ")
                self.attendance_system.mark_attendance(name)
            elif option == "2":
                self.attendance_system.view_attendance()
            elif option == "3":
                name = input("Enter name: ")
                self.attendance_system.view_specific_attendance(name)
            elif option == "4":
                print("Exiting...")
                break
            else:
                print("Invalid option. Please try again.")


# Start the system
attendance_management = AttendanceManagement()
attendance_management.run()

File: attendence.ts
import * as readline from 'readline';
import { format } from 'date-fns';

class AttendanceSystem {
    private attendanceList: { [key: string]: string[] };

    constructor() {
        this.attendanceList = {};
    }

    markAttendance(name: string): void {
        const date = format(new Date(), 'yyyy-MM-dd');

        if (this.attendanceList[name]) {
            if (this.attendanceList[name].includes(date)) {
                console.log(`Attendance already marked for ${name} on ${date}.`);
            } else {
                this.attendanceList[name].push(date);
                console.log(`${name} has been marked present for ${date}.`);
            }
        } else {
            this.attendanceList[name] = [date];
            console.log(`${name} has been marked present for ${date}.`);
        }
    }

    viewAttendance(): void {
        if (Object.keys(this.attendanceList).length === 0) {
            console.log("No attendance records found.");
        } else {
            for (const [name, dates] of Object.entries(this.attendanceList)) {
                console.log(`${name}: ${dates.join(', ')}`);
            }
        }
    }

    viewSpecificAttendance(name: string): void {
        if (this.attendanceList[name]) {
            const dates = this.attendanceList[name];
            console.log(`${name} has been present on: ${dates.join(', ')}`);
        } else {
            console.log(`No attendance records found for ${name}.`);
        }
    }
}

class AttendanceManagement {
    private attendanceSystem = new AttendanceSystem();

    run(): void {
        const rl = readline.createInterface({
            input: process.stdin,
            output: process.stdout
        });

        console.log("Attendance Management System");
        
        while (true) {
            console.log("\nOptions:");
            console.log("1. Mark Attendance");
            console.log("2. View Attendance");
            console.log("3. View Specific Attendance");
            console.log("4. Exit");

            rl.question("Select an option: ", (option) => {
                if (option === "1") {
                    rl.question("Enter name: ", (name) => {
                        this.attendanceSystem.markAttendance(name);
                    });
                } else if (option === "2") {
                    this.attendanceSystem.viewAttendance();
                } else if (option === "3") {
                    rl.question("Enter name: ", (name) => {
                        this.attendanceSystem.viewSpecificAttendance(name);
                    });
                } else if (option === "4") {
                    console.log("Exiting...");
                    rl.close();
                    process.exit(0);
                } else {
                    console.log("Invalid option. Please try again.");
                }
            });
        }
    }
}

// Start the system
const attendanceManagement = new AttendanceManagement();
attendanceManagement.run();
